### PR TITLE
Add repeat_last

### PR DESCRIPTION
**Pull request by d-ryzhikov**
Add `repeat_last(iterable)` function with the following functionality, which after the given *iterable* is exhausted keeps yielding its last element.
```python
>>> list(islice(repeat_last(range(3)), 5))
[0, 1, 2, 2, 2]
```

### CODE REVIEW

File: more_itertools/more.py [hunk 1]
```diff
@@ -1202,6 +1203,26 @@ def padded(iterable, fillvalue=None, n=None, next_multiple=False):
             yield fillvalue
 
 
+def repeat_last(iterable):
```
Comment (bbayles) [inline more_itertools/more.py]: What do you think of having a `default` value to emit endlessly, if *iterable* is empty?

File: more_itertools/more.py [hunk 2]
```diff
@@ -1202,6 +1203,26 @@ def padded(iterable, fillvalue=None, n=None, next_multiple=False):
...
+    """After the *iterable* is exhausted, keep yielding its last element.
+
+        >>> list(islice(repeat_last(range(3)), 5))
+        [0, 1, 2, 2, 2]
+    """
+    it = iter(iterable)
```
Comment (bbayles) [inline more_itertools/more.py]: I think we can simplify this implementation. What about

```python
item = _marker
for item in iterable:
    yield item

final = default if (item is _marker) else item
yield from repeat(final)
```

Comment (d-ryzhikov) [reply more_itertools/more.py]: Didn't notice that there was a marker and didn't want to introduce one. This implementations is much cleaner.

File: more_itertools/more.py [hunk 3]
```diff
@@ -1202,6 +1203,26 @@ def padded(iterable, fillvalue=None, n=None, next_multiple=False):
             yield fillvalue
 
 
+def repeat_last(iterable, default=None):
```
Comment (MSeifert04) [inline more_itertools/more.py]: Shouldn't the default for default be `_marker`?

**Comment by bbayles** [comment]
Excellent - many thanks!

**Pull request merged**